queries.sql

```diff
@@ -1,3 +1,120 @@
+
+/* update species col for animals with names not ending in mon */
+update animals set species = 'pokemon' where species IS null;
+
+/* Confirm changes */
+SELECT * from animals;
+
+/* commit changes */
+COMMIT;
+
+/* Verify that change was made and persists after commit */
+SELECT * from animals;
+
+/* Delete all records in the animals table transaction C */
+/* Begin Transaction C */
+BEGIN;
+
+/* delete all records */
+DELETE from animals;
+
+/* verify all records are deleted */
+SELECT * from animals;
+
+/* Rollback changes */
+ROLLBACK;
+
+/* verify all records still exist */
+SELECT * from animals;
+
+/* Begin Transaction D */
+BEGIN;
+
+/* Delete all animals born after Jan 1st, 2022. */
+DELETE from animals WHERE date_of_birth > '01-01-2022';
+
+/* create a savepoint */
+SAVEPOINT dob;
+
+/* update all animals' weights multiplied by -1 */
+UPDATE animals SET weight_kg =weight_kg * -1;
+
+/* roolback to savepoint */
+ROLLBACK TO dob;
+
+/* update all animals' weights that are negative multiplied by -1 */
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
+/* commit the transaction */
+COMMIT;
+
+/* How many animals are there? */
+SELECT COUNT(*) from animals;
+
+/* How many animals have never tried to escape? */
+SELECT COUNT(*) from animals WHERE escape_attempts = 0;
+
+/* What is the average weight of animals? */
+SELECT AVG(weight_kg) from animals;
+
+/* Who escapes the most, neutered or not neutered animals? */
+SELECT neutered, count(neutered) as result FROM animals GROUP BY neutered ORDER BY result DESC LIMIT 1;
+
+/* What is the minimum and maximum weight of each type of animal? */
+SELECT species, MAX(weight_kg) AS max_weight, MIN(weight_kg) AS min_weight FROM animals GROUP BY species;
+
+/* What is the average number of escape attempts per animal type of those born between 1990 and 2000? */
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
+
+/* What animals belong to Melody Pond? */
+SELECT name, full_name FROM animals JOIN owners ON animals.owner_id = owners.id WHERE full_name = 'Melody Pond';
+
+/* List animals that are pokemon. */
+SELECT animals.name, species.name FROM animals JOIN species ON animals.species_id = species.id WHERE species.name = 'Pokemon';
+
+/* List of owners and their animals, remember to include those that don't own any animal. */
+SELECT name, full_name FROM animals FULL JOIN owners ON animals.owner_id = owners.id;
+
+/* How many animals are there per species? */
+SELECT COUNT(animals.name), species.name FROM animals JOIN species ON animals.species_id = species.id GROUP BY species.name;
+
+/* List all Digimon owned by Jennifer Orwell. */
+SELECT owners.full_name, animals.name AS animal_name, species.name AS species_name FROM animals JOIN owners ON owners.id = animals.owner_id JOIN species ON species.id = animals.species_id WHERE owners.full_name = 'Jennifer Orwell' AND species.name = 'Digimon';
```

Comment:
- [ ] It would be better if we take a second look at the results of this query...Currently, we can not find any data as a result of the query, however, we are expecting to find a data from this query. Kindly check the screenshot to this query